src/shared/dropdown-list/dropdown-list.tsx

```diff
@@ -205,18 +205,9 @@ export const DropdownList = <T, V>({
                     typeof label === "object"
                         ? label.title.toLowerCase()
                         : label.toLowerCase();
+                const secondaryLabel = typeof label === "string" ? undefined : label.secondaryLabel.toLowerCase();;
```

Comment:
based on the types, secondaryLabel is optional so we should also check it exists first
```suggestion
                const secondaryLabel = typeof label === "string" ? undefined : label.secondaryLabel?.toLowerCase();
```